src/reducers/endpointExplorer.js

```diff
@@ -79,15 +75,39 @@ function pendingRequestValues(state={}, action) {
   return state;
 }
 
-function currentRequest(state={isLoading: false}, action) {
-  switch (action.type) {
-  case START_REQUEST:
-    return Object.assign({}, state, {isLoading: true});
-  case FINISH_REQUEST:
-    let {error, payload} = action;
+function results(state={id: null, available: false, body: []}, action) {
+  if (action.type === START_REQUEST) {
+    return Object.assign({}, state, {
+      available: true,
+      id: action.id,
+      isError: false,
+      body: [],
+    });
+  }
 
-    return Object.assign({}, state, {isLoading: false, response: payload, error});
-  default:
+  if (action.id !== state.id) {
+    // This action has expired as we've moved on to a new request
+    // Or, this is likely an irrelevant action
     return state;
   }
+
+  if (action.type === ERROR_REQUEST) {
+    let errorContent = action.errorStatus === 0 ?
```

Comment:
IMO you're getting a little fancy with this statement... ternary expressions split onto multiple lines makes me think you should just use a regular old if/else block.